codec/src/main/java/io/netty/handler/codec/protobuf/ProtobufVarint32FrameDecoder.java

```diff
@@ -70,52 +70,68 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out)
      *
      * @return decoded int if buffers readerIndex has been forwarded else nonsense value
      */
-    private static int readRawVarint32(ByteBuf buffer) {
+    static int readRawVarint32(ByteBuf buffer) {
+        if (buffer.readableBytes() < 4) {
+            return readRawVarint24(buffer);
+        }
+        int wholeOrMore = buffer.getIntLE(buffer.readerIndex());
+        int firstOneOnStop = ~wholeOrMore & 0x80808080;
+        if (firstOneOnStop == 0) {
+            return readRawVarint40(buffer, wholeOrMore);
+        }
+        int bitsToKeep = Integer.numberOfTrailingZeros(firstOneOnStop) + 1;
+        buffer.skipBytes(bitsToKeep >> 3);
```

Comment:
I've noticed something very fun here: if I use `readerIndex(int)` it get slightly slower :( 
which is very weird, because (FYI @chrisvest ), `skipBytes` check accessibility, while `readerIndex(int)`, nope!